src/video/SDL_video.c

```diff
@@ -763,29 +767,55 @@ SDL_DisplayID SDL_AddVideoDisplay(const SDL_VideoDisplay *display, SDL_bool send
         new_display->fullscreen_modes[i].displayID = id;
     }
 
-    props = SDL_GetDisplayProperties(id);
-
-    if (display->HDR.HDR_headroom > 1.0f) {
-        SDL_SetBooleanProperty(props, SDL_PROP_DISPLAY_HDR_ENABLED_BOOLEAN, SDL_TRUE);
-    } else {
-        SDL_SetBooleanProperty(props, SDL_PROP_DISPLAY_HDR_ENABLED_BOOLEAN, SDL_FALSE);
+    SDL_copyp(&new_display->HDR, &display->HDR);
+    if (new_display->HDR.SDR_white_level < 1.0f) {
+        new_display->HDR.SDR_white_level = 1.0f;
     }
-    if (display->HDR.SDR_white_point <= 1.0f) {
-        SDL_SetFloatProperty(props, SDL_PROP_DISPLAY_SDR_WHITE_POINT_FLOAT, 1.0f);
-    } else {
-        SDL_SetFloatProperty(props, SDL_PROP_DISPLAY_SDR_WHITE_POINT_FLOAT, display->HDR.SDR_white_point);
-    }
-    if (display->HDR.HDR_headroom <= 1.0f) {
-        SDL_SetFloatProperty(props, SDL_PROP_DISPLAY_HDR_HEADROOM_FLOAT, 1.0f);
-    } else {
-        SDL_SetFloatProperty(props, SDL_PROP_DISPLAY_HDR_HEADROOM_FLOAT, display->HDR.HDR_headroom);
+    if (new_display->HDR.HDR_headroom < 1.0f) {
+        new_display->HDR.HDR_headroom = 1.0f;
     }
 
     SDL_UpdateDesktopBounds();
 
     return id;
 }
 
+void SDL_SetDisplayHDRProperties(SDL_VideoDisplay *display, const SDL_HDRProperties *HDR)
+{
+    SDL_bool changed = SDL_memcmp(HDR, &display->HDR, sizeof(SDL_HDRProperties)) != 0;
+
+    if (changed) {
+        SDL_copyp(&display->HDR, HDR);
+        if (display->HDR.SDR_white_level < 1.0f) {
+            display->HDR.SDR_white_level = 1.0f;
+        }
+        if (display->HDR.HDR_headroom < 1.0f) {
+            display->HDR.HDR_headroom = 1.0f;
+        }
+
+        SDL_SendDisplayEvent(display, SDL_EVENT_WINDOW_HDR_STATE_CHANGED, 0);
```

Comment:
The HDR state change isn't a display event. We should either have a distinct event for displays or only send it to windows.